app/javascript/packs/application/App.js

```diff
@@ -1,112 +1,52 @@
 import api from "./api";
-
-const INITIAL_STATE = {
-  currentStaffUser: null,
-  tours: {
-    data: null,
-    isFetching: false
-  },
-  tour: {
-    data: null,
-    isFetching: false
-  },
-  points: {
-    data: null,
-    isFetching: false
-  }
-};
+import * as staffUserActions from "./redux/staffUser/actions";
+import * as staffUserSelectors from "./redux/staffUser/selectors";
 
 class App extends Component {
-  state = INITIAL_STATE;
-
   componentDidMount() {
-    this.authenticateStaffUser();
+    this.props.authenticateStaffUser();
   }
 
-  authenticateStaffUser = async () => {
-    const result = await api.authenticateStaffUser();
+  componentDidUpdate(prevProps) {
+    const wasLoggedIn = !!prevProps.currentStaffUser;
+    const isLoggedIn = !!this.props.currentStaffUser;
 
-    if (result.ok) {
-      this.setState({ currentStaffUser: result.data });
+    if (wasLoggedIn === isLoggedIn) {
+      return;
     }
 
-    return result;
-  };
-
-  loginStaffUser = async (username, password) => {
-    const result = await api.login(username, password);
-
-    if (result.ok) {
-      return this.authenticateStaffUser();
+    if (isLoggedIn) {
+      return navigate("/admin");
     }
 
-    return result;
-  };
-
-  logoutStaffUser = event => {
-    event.preventDefault();
-    api.removeAuthToken();
-
-    this.setState(INITIAL_STATE, () => navigate("/login"));
-  };
-
-  registerStaffUser = async attributes => {
-    const userResult = await api.createStaffUser(attributes);
-    const { username, password } = attributes.user;
-
-    if (userResult.ok) {
-      return this.loginStaffUser(username, password);
+    if (wasLoggedIn) {
+      return navigate("/login");
     }
+  }
 
-    return userResult;
-  };
-
-  registerTour = async attributes => {
-    const tourResult = await api.createTour(attributes);
-    return tourResult;
-  };
-
-  listTours = async () => {
-    this.setState({ tours: { isFetching: true, data: null } });
-    const tourList = await api.listTours();
-    this.setState({ tours: { isFetching: false, data: tourList.data } });
-  };
-
-  showTour = async tourId => {
-    this.setState({ tour: { isFetching: true, data: null } });
-    const tour = await api.getTour(tourId);
-    this.setState({ tour: { isFetching: false, data: tour.data } });
-  };
-
-  registerPoint = async attributes => {
-    const tourId = this.state.tour.data.id;
-    const pointResult = await api.createPoint(tourId, attributes);
-    return pointResult;
-  };
-
-  listPoints = async tourId => {
-    this.setState({ points: { isFetching: true, data: null } });
-    const pointList = await api.listPoints(tourId);
-    this.setState({ points: { isFetching: false, data: pointList.data } });
+  handleLogout = event => {
+    event.preventDefault();
+    this.props.logoutStaffUser();
   };
 
   render() {
     return (
       <div>
         <h1>Tour-est</h1>
-        {this.state.currentStaffUser ? (
+        {this.props.currentStaffUser ? (
```

Comment:
I think your code would be easier to read if you extract this conditional display logic into its own little component.  Then you could test it easily 😎